x-pack/plugins/cases/public/components/system_actions/cases/cases_params.tsx

```diff
@@ -0,0 +1,213 @@
+    (key: 'timeWindowSize' | 'timeWindowUnit', value: string) => {
+      if (!value) {
+        return;
+      }
+
+      const newTimeWindow =
+        key === 'timeWindowSize' ? `${value}${timeWindowUnit}` : `${timeWindowSize}${value}`;
+
+      editSubActionProperty('timeWindow', newTimeWindow);
+    },
+    [editSubActionProperty, timeWindowUnit, timeWindowSize]
+  );
+
+  const onChangeComboBox = useCallback(
+    (optionsValue: Array<EuiComboBoxOptionOption<string>>) => {
+      editSubActionProperty('groupingBy', optionsValue?.length ? [optionsValue[0].value] : []);
+    },
+    [editSubActionProperty]
+  );
+
+  const options: Array<EuiComboBoxOptionOption<string>> = useMemo(() => {
+    if (!dataViews?.length) {
+      return [];
+    }
+
+    return dataViews
+      .map((dataView) => {
+        return dataView.fields
+          .filter((field) => field.esTypes?.includes('keyword'))
+          .map((field) => ({
+            value: field.name,
+            label: field.name,
+          }));
+      })
+      .flat();
+  }, [dataViews]);
+
+  const selectedOptions = groupingBy.map((field) => ({ value: field, label: field }));
+
+  return (
+    <>
+      <EuiFlexGroup>
+        <EuiFlexItem grow={true}>
+          <EuiFormRow fullWidth>
+            <EuiComboBox
+              fullWidth
+              isClearable={true}
+              singleSelection
+              data-test-subj="group-by-alert-field-combobox"
+              prepend={i18n.GROUP_BY_ALERT}
+              isLoading={loadingAlertDataViews}
+              isDisabled={loadingAlertDataViews}
+              options={options}
+              onChange={onChangeComboBox}
+              selectedOptions={selectedOptions}
+            />
+          </EuiFormRow>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+      <EuiSpacer size="m" />
+      <EuiFormRow
+        fullWidth
+        id="timeWindow"
+        error={errors.timeWindow}
+        isInvalid={
+          errors.timeWindow !== undefined &&
+          errors.timeWindow.length > 0 &&
+          timeWindow !== undefined
+        }
+      >
+        <EuiFlexGroup alignItems="flexEnd" gutterSize="s">
+          <EuiFlexItem grow={4}>
+            <EuiFieldNumber
+              prepend={i18n.TIME_WINDOW}
+              data-test-subj="time-window-size-input"
+              value={timeWindowSize}
+              min={1}
+              step={1}
+              onChange={(e) => {
+                handleTimeWindowChange('timeWindowSize', e.target.value);
+              }}
+            />
+          </EuiFlexItem>
+          <EuiFlexItem grow={3}>
+            <EuiSelect
+              fullWidth
+              data-test-subj="time-window-unit-select"
+              value={timeWindowUnit}
+              onChange={(e) => {
+                handleTimeWindowChange('timeWindowUnit', e.target.value);
+              }}
+              options={getTimeUnitOptions(timeWindowSize ? parseInt(timeWindowSize, 10) : 1)}
```

Comment:
A reminder that `parseInt` can return `NaN`. In `getTimeUnitOptions` you converted again to string so maybe it is not needed to make it an integer here.

Reply:
removed parseInt in https://github.com/elastic/kibana/pull/179409/commits/c67258238cf90cecb683a14ebd6998e22aa175c6